app.py

```diff
@@ -233,6 +246,15 @@ def generate_select_boxes(options_x, options_y, options_color, key_prefix):
                     col_color=col_color,
                     agg="mean",
                 )
+                st.header("Heatmap series plot")
+                plot_series_heatmap(
+                    df,
+                    col_x=col_x,
+                    col_y=col_y,
```

Comment:
Instead of col_x and col_y, I think we would only need to define col_date and col_color. Both X and Y axis are taken from col_date but at different time units

Reply:
Yea I did think this while coding this up. so col_x as you know by default is "date", now for the X and Y axis, I have used Time Scale (i.e Day v/s month, month v/s year etc..). Now, for the colour, I have used the entry of col_y. And then I do have the aggregation method which specifies what to use for aggregating the col_y data which specifies the colour of the heatmap.